[16.0] date_range lost compatibility with web_advanced_search

<!-- Provide a general summary of the issue in the Title above.
Before opening the issue, please check that there isn't another issue addressing the same problem already created -->

## Module
date_range

## Describe the bug
After https://github.com/OCA/server-ux/pull/717 web_advanced_search is not working properly, throws this error

```
OwlError: An error occured in the owl lifecycle (see this Error's "cause" property)
    at handleError (https://XXXXXXXXXXXXXXXXXX/web/assets/1122-191e1e2/web.assets_common.min.js:1472:101)
    at owl.App.handleError (https://XXXXXXXXXXXXXXXXXX/web/assets/1122-191e1e2/web.assets_common.min.js:2098:29)
    at RootFiber._render (https://XXXXXXXXXXXXXXXXXX/web/assets/1122-191e1e2/web.assets_common.min.js:1497:19)
    at RootFiber.render (https://XXXXXXXXXXXXXXXXXX/web/assets/1122-191e1e2/web.assets_common.min.js:1495:6)
    at ComponentNode.render (https://XXXXXXXXXXXXXXXXXX/web/assets/1122-191e1e2/web.assets_common.min.js:1571:56)

Caused by: TypeError: Cannot read properties of undefined (reading '0')
    at CustomFilterItem.setDefaultValue (https://XXXXXXXXXXXXXXXXXX/web/assets/1167-8cb4da3/web.assets_backend.min.js:1837:154)
    at CustomFilterItem.setDefaultValue (https://XXXXXXXXXXXXXXXXXX/web/assets/1167-8cb4da3/web.assets_backend.min.js:15435:459)
    at setDefaultValue (patch date_range.CustomFilterItem) (https://XXXXXXXXXXXXXXXXXX/web/assets/1167-8cb4da3/web.assets_backend.min.js:1657:22)
    at CustomFilterItem.setDefaultValue (https://XXXXXXXXXXXXXXXXXX/web/assets/1167-8cb4da3/web.assets_backend.min.js:15235:579)
    at setDefaultValue (patch web_advanced_search.CustomFilterItem) (https://XXXXXXXXXXXXXXXXXX/web/assets/1167-8cb4da3/web.assets_backend.min.js:1657:22)
    at CustomFilterItem.addNewCondition (https://XXXXXXXXXXXXXXXXXX/web/assets/1167-8cb4da3/web.assets_backend.min.js:1835:155)
    at CustomFilterItem.setup (https://XXXXXXXXXXXXXXXXXX/web/assets/1167-8cb4da3/web.assets_backend.min.js:1834:392)
    at CustomFilterItem.setup (https://XXXXXXXXXXXXXXXXXX/web/assets/1167-8cb4da3/web.assets_backend.min.js:15434:484)
    at setup (patch date_range.CustomFilterItem) (https://XXXXXXXXXXXXXXXXXX/web/assets/1167-8cb4da3/web.assets_backend.min.js:1657:22)
    at CustomFilterItem.setup (https://XXXXXXXXXXXXXXXXXX/web/assets/1167-8cb4da3/web.assets_backend.min.js:15235:421)
```

## To Reproduce
**Affected versions**:

Steps to reproduce the behavior:
1. with date_range and web_advanced_search installed 
2. On valuation report (stock_account)  Inventory> Reporting>Valuation (similar response could have with other model )
3. Click inside "Filters" and move your mouse between "Group By" and "Filters"


https://github.com/OCA/server-ux/assets/4610000/1adf4f96-775c-470d-a79f-8cfe954ee420

CC @CarlosRoca13

--- a/date_range/static/src/js/date_range.esm.js
+++ b/date_range/static/src/js/date_range.esm.js
@@ -1,6 +1,6 @@
 /** @odoo-module **/
-import {FIELD_OPERATORS, FIELD_TYPES} from "web.searchUtils";
 import {CustomFilterItem} from "@web/search/filter_menu/custom_filter_item";
+import {FIELD_TYPES} from "web.searchUtils";
 import {_lt} from "@web/core/l10n/translation";
 import {patch} from "@web/core/utils/patch";
 import {useService} from "@web/core/utils/hooks";
@@ -15,9 +15,6 @@ patch(CustomFilterItem.prototype, "date_range.CustomFilterItem", {
     },
 
     async _computeDateRangeOperators() {
-        this.OPERATORS = Object.assign({}, FIELD_OPERATORS);
-        this.OPERATORS.date = [...FIELD_OPERATORS.date];
-        this.OPERATORS.datetime = [...FIELD_OPERATORS.datetime];
         this.date_ranges = {};
         const result = await this.orm.searchRead(
             "date.range",

